Prevents shoving while crawling

Bm0n opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## What Does This PR Do
<!-- Include a small to medium description of what your PR changes. -->
<!-- Document all changes, as not doing this may delay reviews or even discourage maintainers from merging your PR! -->
<!-- If your PR fixes an issue, add "Fixes #1234" somewhere in the PR description. This will automatically close the bug upon PR submission. -->
Adds a simple check to prevent shoving while crawling or resting.

This only changes shoving, you can still attack.
## Why It's Good For The Game
<!-- Add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->
Somebody care to explain to me how you're able to shove someone else into a wall/person while you yourself are splayed out on the floor?

You can't and shouldn't be able to. This change reflects how shoving works on other servers that have it and is how shoving should behave here on Paradise too. I've seen enough people rest, crawl under someone then to one tile adjacent and shove them into someone else to know this is something that needs to be changed.

Probably going to work on kicking next to really cement home the idea crawling around everywhere on purpose isn't a good idea, but that's for another PR.

## Testing
<!-- How did you test the PR, if at all? -->
In game.
## Changelog
:cl: Bmon
tweak: You can no longer shove while crawling.
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->
<!-- If a PR has no impact on players (i.e. a code refactor that does not change functionality) then the entire Changelog heading and contents can be removed. -->

Bm0n commented:
> The point of changing stuns to knockdowns was to give people more of a chance to fight back, wasn't it? This goes against the spirit of that change, as I see it. 

Shoving was added on both TG and Para to remove the RNG from disarms, not to give people more of a chance to fight back. I don't know where you got that idea from. 

> Shoving while crawling isn't just the meme you're portraying it as. It legitimately gives people chances to defend themselves when they otherwise wouldn't.

Nothing is stopping you from pulling a baton, throwing a handtele portal down and climbing into it, crushing a bluespace crystal in your hands or crawling under the person who just shoved you. All of which good counters to a shove.

> Realism does not necessarily improve the game.

To an extent you're right, but this is flat-out silly. It is physically impossible to impart your body weight onto someone else shoving them while all of it is resting on the floor. Still, I see this as more of a balance issue than a realism one.

Komrad822 commented:
So I see the benefits and disadvantages of this in the eyes of an antag. As cult, you stun hand and then need to stab the target with your dagger OR have another stun hand ready. The target is knocked down and can still shove you, spam it even so that each time you approach you are prevented from doing the finishing blow. This is highly infuriating since the victim gains the advantage here unless it was completely by surprise (you have to approach THEM, thus they dictate the engage distance and can spam click you as you approach to get the guaranteed shove). This applies to contractor baton too.

But on another note, this gives the antag an advantage when dealing with sec in terms of snatching their baton up and turning the fight without having to use antistun abilities. Shoving can make or break an engagement, especially if you're able to get a good shove onto a table for the stun time.

I wouldnt be against cooldown on the shove if you're on the ground. This gives you an opportunity to resist but you cant stave off the person forever.

Bm0n commented:
> But on another note, this gives the antag an advantage when dealing with sec in terms of snatching their baton up and turning the fight without having to use antistun abilities. Shoving can make or break an engagement, especially if you're able to get a good shove onto a table for the stun time.

If you get shoved because of your own poor positioning by someone you know is charging you that's on you and shouldn't be able to tit for tat shove them back until you first get back up on your feet. That's not to say you're helpless if you get shoved, as a secoff there are many things you can do to counter this; flashbangs, flash, disabler, the rest of security which you should be patrolling with, all of which can help you if you get shoved. You're not helpless, this is a poor notion in my opinion.

Also, if you're patrolling around with your baton out in your hands with zero intention of using it... yikes. Maybe don't do that. Even before this PR it was a bad idea to do that, don't walk around with your weapon out unless you intend on using it.

> I wouldnt be against cooldown on the shove if you're on the ground. This gives you an opportunity to resist but you cant stave off the person forever.

I am fully against adding a cooldown. You shouldn't be able to shove while resting and I think I have done a good job explaining why.

Carthusia commented:
I dunno, i feel removing this would basicly destroy our current "Donnybrook" fighting style, where life and death struggles actually occur, instead of "Stun, dead" and "Knockdown, keep down, dead"

People have a chance to fight back now, and whether that was intended or not, thats whats happened and its actually pretty nice and far cry better than the old system.

Not a fan of the idea of putting us back towards the days of near 1 hit stuns

hal9000PR commented on code/modules/mob/living/carbon/human/species/_species.dm:
```diff
@@ -515,6 +515,8 @@
 /datum/species/proc/disarm(mob/living/carbon/human/user, mob/living/carbon/human/target, datum/martial_art/attacker_style)
 	if(user == target)
 		return FALSE
+	if(IS_HORIZONTAL(user))
+		return FALSE
```
this will prevent people who have martial arts from doing combos involving disarms while downed aswell.

Bm0n replied:
should be fixed now

Komrad822 commented:
>I am fully against adding a cooldown. You shouldn't be able to shove while resting and I think I have done a good job explaining why.

Your current explanations are:
1. Other servers dont do this
2. It doesnt make sense irl

While yes you are on the floor and you limit your application of force, this is also a game where there are space wizards, an infinite power generating glowing nacho, and space birds that hate oxygen. Ive stated why removing shoves would help the aggressor as it removes the victims ability to fight back by spamming shove which knocks the aggressor back, knocks them down if against a wall, and disarms on the second shove. But I also stated that it makes close fights interesting. If someone is going to misplay by allowing themselves to get shoved by a person on the ground, they should be punished for it.

We moved away from irl combat ideas by removing instant stuns, tasers are irl, they can take down a full grown human unless they are on stimulants (Florida man on bath salts moment). Slugs blow holes the size of a baseball in human beings, But we removed both of those because they are unfun. Yes shoves are not in TG while you are down but I know some servers it takes more than 2 baton hits and 6 disabler hits to take someone down. Shoves help the defenseless fight back, whether they are regular crew, a striped officer, or a striped antag. 

>Nothing is stopping you from pulling a baton, throwing a handtele portal down and climbing into it, crushing a bluespace crystal in your hands or crawling under the person who just shoved you. All of which good counters to a shove.

Not everyone is armed or ready to be taken victim. Theres only two hand tele's even then on Para the portal stays open. Tele baton is an instant down atm, its counter is shoving the person. Blue space crystals have been nerfed and not everyone has these nor should they be required. Crawling under the person doesn't stop them from getting the second baton off, removing their tool does. 

Again, I dont like being spam shoved by the person on the ground but I also dont want to be a defenseless fish on the ground who got hit once by a down (glare, baton, stun hand, telebaton) and only have my fist.

>Shoving was added on both TG and Para to remove the RNG from disarms, not to give people more of a chance to fight back. I don't know where you got that idea from.

May not have been the intention but its currently how its used. We moved away from stun combat so its not one click win, shoves are currently a great way to make it so you aren't just on the floor waiting for your inevitable end as you are about to be stam crit.

Bm0n commented:
>Not everyone is armed or ready to be taken victim. Theres only two hand tele's even then on Para the portal stays open. Tele baton is an instant down atm, its counter is shoving the person. Blue space crystals have been nerfed and not everyone has these nor should they be required. Crawling under the person doesn't stop them from getting the second baton off, removing their tool does.

>Again, I dont like being spam shoved by the person on the ground but I also dont want to be a defenseless fish on the ground who got hit once by a down (glare, baton, stun hand, telebaton) and only have my fist.

If you have nothing literally, crawl under the person who shoved you. You get up faster than they can reposition and shove you again. If it was a baton and not a shove oh well you fucked anyways unless you have an anti-stun. That's how it should work and how our combat is balanced here on Paradise. Almost every antag has an anti-stun for this very reason and antags usually have an advantage over crew whether you like it or not.

>We moved away from irl combat ideas by removing instant stuns, tasers are irl, they can take down a full grown human unless they are on stimulants (Florida man on bath salts moment). Slugs blow holes the size of a baseball in human beings, But we removed both of those because they are unfun to deal with every single round as they were so easy to get. 

Much like everything you just listed it is unfun to get shoved by someone on the floor crawling around.

>Ive stated why removing shoves would help the aggressor as it removes the victims ability to fight back by spamming shove which knocks the aggressor back, knocks them down if against a wall, and disarms on the second shove. But I also stated that it makes close fights interesting. If someone is going to misplay by allowing themselves to get shoved by a person on the ground, they should be punished for it.

Nothing is interesting about mindlessly spamming shoves after you get batoned, there is zero positioning involved and zero skill involved, just pure spam.


> Your current explanations are:
> 
> 1. Other servers dont do this
> 2. It doesnt make sense irl

>While yes you are on the floor and you limit your application of force, this is also a game where there are space wizards, an infinite power generating glowing nacho, and space birds that hate oxygen.

It's like you're ignoring half of what I am saying and picking and choosing what you want to hear. I refuse to argue this with you, go back and re-read what I said because I didn't list only two things.

![a](https://user-images.githubusercontent.com/92271472/207422606-ef700383-2ee6-48b1-bb29-e9ea83f3957b.png)

Bm0n commented:
>whatever komrad said

I said I was done I meant it, I am not reading any of this.

Qwertytoforty commented:
I like the idea of the PR, however I would be tempted to make disarm intent while lying down still push people back, so it has some impact still, without pushing back with enough force to knock onto a table / knock down against a wall.
This way people at least have some feeling of having an effect, vs a shove doing nothing.

Bm0n commented:
> I like the idea of the PR, however I would be tempted to make disarm intent while lying down still push people back, so it has some impact still, without pushing back with enough force to knock onto a table / knock down against a wall. This way people at least have some feeling of having an effect, vs a shove doing nothing.

I agree that's probably the right compromise over adding a timer but mannn I honestly think you shouldn't be able to shove when rested in the first place. It's borked to all hell that you can in the first place.

I'll do it if one of the balance team members makes it a requirement for merge.

OctusGit commented:
I'm pretty much 100% for this PR- I find a lot of times that being on the floor is often too forgiving for something that should be rather punishing as a mechanic. 

Spam shoving is probably one of the most common things for people to throw and when your floored you actually have a much better advantage at positioning because of being able to freely move under people.

While we could throw out the idea of not being able to crawl under people, I still find that shoving while being floored seems a lot easier to do, and frankly I find it rather inconsistent for a game mechanic that is supposed to punish.

Mitchs98 commented:
Disagreeing with this PR for reasons that Komrad and Carthusia have already stated. I understand that it's this way on /tg/ and other servers...but I've said it before and I'll say it again. We are not them, we do not have to mirror them just because that's what they do.

As annoying as it is to be shoved in a cell by a prisoner and have to fight them if you don't properly stun them...I still disagree. If you're knocked down, as Komrad stated, your only option is hoping you have some kind of weapon or you can shove your attacker away. It's not realistic, but otherwise you might as well just have been hit with an instant stun because more often than not the antagonist will certainly have a way to just instantly stun you afterwards. 

It's not fun, at all, and that was the point of disarm change and moving to stamina combat; to make combat more fun and involved other than walking up to Joe McBaldman and instantly batoning/prodding/tasing him into oblivion.

People *should* be punished for not positioning themselves properly in a fight, even if it ends up being from the person they just stunned.

There's a difference between 'Antagonists should always have advantages over crew.' as opposed to 'You may as well tab out any time you get stunned and do not have a weapon.'. Granted, in most situations you WILL still lose regardless even with shoving. It's still far more engaging to have that chance that you might be able to ward off your attacker and get away than it is for you to be able to do nothing really at all.

I just don't see a reason for this outside of 'Realism', 'I don't like this', and 'TG does this', to be blunt. Not one that's truly meant to be making combat more fun, engaging, and interesting like the original changes were meant to do.

Even Qwerty's compromise would be better than the PR the state is currently in.

Bm0n commented:
> I just don't see a reason for this outside of 'Realism', 'I don't like this', and 'TG does this', to be blunt. Not one that's truly meant to be making combat more fun, engaging, and interesting like the original changes were meant to do.

you know I think I am going insane. it's that or people are just not listening

should have never mentioned other servers at all. specifically didn't mention TG on purpose and yet the conversation still ends up there

matttheficus commented:
I'm with Qwerty on this one. If your ass is on the ground at MOST youre going to be trying to push someone away with your legs, so having a push-back would be nice. Removing the ability to wall/table slam someone while prone would be a good compromise.

procdrone commented:
i agree with qwerty way, just remove the knockdown while disarming prone.

Bm0n commented:
I am sticking to my guns and not changing anything unless it's a requirement for merge. 

Why? A gimped shove is completely pointless, you're honestly better off doing anything else if you get knocked down than trying to shove the person who knocked you down with a shove that does nothing.

Also, people can still be annoying ass floor rats with it and shove people around(even if it does nothing) so egh pass for me.

procdrone commented:
> Why? A gimped shove is completely pointless, you're honestly better off doing anything else if you get knocked down than trying to shove the person who knocked you down with a shove that does nothing.

A person that is trying to run away from an attacker can successfully push people away and prevent melee attacks, or at least, stop some of them, giving themselves more time to get away/crawl further away (for example, out of maints, or into a major area to get help from others)

Bm0n commented:
> A person that is trying to run away from an attacker can successfully push people away and prevent melee attacks, or at least, stop some of them, giving themselves more time to get away/crawl further away (for example, out of maints, or into a major area to get help from others)

A shove that imparts zero slow down, zero knockdown and only moves you one tile away is very, very, very, very negligible.

The person who is attacking you, standing upright, moving THREE TIMES as fast as you crawling on the floor would just move one or two tiles over and continue to attack you. God forbid your attacker is up against a wall where your newly gimped shove will actually do nothing.

Let's be honest here, the only reason shoves are useful is because they can impart knockdowns and slowdowns. You're not getting away from anyone by moving them one tile away from you for less than a second, especially since they're moving(i really want to hammer this home) three times faster than you.

This is why I am against adding a gimped shove and believe that we should just completely disable it for people in a knocked down or crawling state. I'd be adding a feature that'd only useful purpose would be to annoy people with and be utterly useless in actual combat.

Peakwastaken commented:
I'd very much like to keep horizontal shoving with the slowdown it gives, and the two shove disarms (easily preventable if you're careful/switch hand/don't miss a click) so there's at least SOMETHING you can do if you don't have any extra equipment (which all of crew shouldn't have besides antag/sec), without the ability to wall/table knockdown.

Bm0n commented:
> I'd very much like to keep horizontal shoving with the slowdown it gives, and the two shove disarms (easily preventable if you're careful/switch hand/don't miss a click) so there's at least SOMETHING you can do if you don't have any extra equipment (which all of crew shouldn't have besides antag/sec), without the ability to wall/table knockdown.

No, definitely not keeping the disarm while slowed down thing. Again, defeating the purpose of this PR.

It was actually something else I wanted to gut from our shoves considering it's literally old disarms but predictable and better, but that's another can of worms that I am not getting into for another time.

Actually hilarious to me that you can game it by switching hands, that was a bug I was unaware of. ALSO, YOU CLICK TO SWITCH HANDS?!

Peakwastaken commented:
To give a bit more depths to my thought because I think it's an important and interesting discussion we're having :

the two shove disarm knocks what's in your active hand down, hence why switching hand prevents it, and it does it after the shove, putting some distance between you and the horizontal person so it's really not that easy for them to turn around and nab it, especially if they were already going the other direction - as they probably would be.

>Shoving was added on both TG and Para to remove the RNG from disarms, not to give people more of a chance to fight back. I don't know where you got that idea from.

The point of disarms is to be able to fight back without a weapon, the point of removing stun combat was to give people more chances to fight back, if they're unable to do the former as soon as the mechanics from the latter enter play, then I do feel we're reverting some of the progress we've made in our combat system.

>defeating the purpose of this PR.

Then what exactly is the point of the PR ? I really struggle to find a straight answer to that question from your OP or answers in the comment

I totally agree that being the first one to fall should definitely put you at a disadvantage, but without the middle ground we're proposing, you're killing most of the back and forth any combat may have for 90% of the crew. Speed is one of the most important aspects of the game and being kd'd slowed PLUS unable to potentially one tap kd/table someone else, instead requiring at least two shoves (again, only for the other person to drop their weapon, usually out of reach, which they can immediately pick back up) I think is pretty fair for the person who's in the superior position. 

>It is physically impossible to impart your body weight onto someone else shoving them while all of it is resting on the floor. Still, I see this as more of a balance issue than a realism one.

As someone very much into bjj/mma I wholly disagree with this

Bm0n commented:
>the two shove disarm knocks what's in your active hand down, hence why switching hand prevents it, and it does it AFTER the knockback, putting some distance between you and the horizontal person so it's really not that easy for them to turn around and nab it, especially if they were already going the other direction - as they probably would be.

Yes, I read the code before posting that, no idea why it's targeting the active hand for the disarm, it's probably an oversight or in other words a bug.

>The point of disarms is to be able to fight back without a weapon, the point of removing stun combat was to give people more chances to fight back and make it more interesting gameplay, if they're unable to do the former as soon as the mechanics from the latter enter play, then I do feel we're reverting some of the progress we've made in our combat system.

And we replaced it with a more powerful version of what we had before, this is the root issue here. I am taking away stuns in this PR not adding them. What do you fail to see? Nothing about floor shoving is interesting or fun because-

>Then what exactly is the point of the PR and your gameplay-impact reasoning behind it? 

-Let me really blunt with you. It's not fun to fight someone and have them do 180 on you and knock you down and then you get fucked by them and their teammate because they imparted a knockdown on you with a flick shove from a semi-stunned state. That's not fun, it put the leverage even more so on who has the most people and that's usually sec.

Yes, they could pull a baton out or try to grab the baton they probably already had out to do the same thing but it's an action that requires effort and focus to do, unlike a shove which is just reactionary spam clicking.

This is the last time I repeat myself. Disarms are in a completely unbalanced state right now and need to be toned back, this is one step towards that.

There are many things that make Paradise shoves way more potent than they should be, in fact, old disarms are arguably worse than what we have now, an overly buffed version of TG shoves. This shouldn't have ever been the case, it doesn't work I seek to change that by rebalancing shoves starting with shoving while laying down. Probably one of if not the most egregiously broken things, in my opinion, about our version of shoves. 

>As someone very much into bjj/mma I wholly disagree with this, and I hope you understand my reasoning balance wise

the average person is not an MMA fighter, I did add a code exception for carp/cqc users.

hal9000PR commented:
I am  gonna object to this.

I feel like being floored could have more of a downside, maybe an innate attackmove mod, however I am not sure I agree with this. 
One idea would be to make getting floored set your next attack move to something like 1-ish second. so you always should lose the drop, steal item fight.

anyway, this PR has 3 balance objections and will be filed as closed.

Bm0n commented:
Very disheartening outcome from the balance team. Would have thought they'd see eye to eye with our shoves being in an untenable state.
